quipucords/api/inspectresult/model.py

```diff
@@ -85,7 +85,7 @@ class RawFactEncoder(JSONEncoder):
 
     def default(self, o):
         """Update the default Encoder to handle types beyond just the basic ones."""
-        if isinstance(o, BaseModel):
+        if isinstance(o, (BaseModel, PydanticErrorProxy)):
```

Comment:
What exactly does adding `PydanticErrorProxy` do for us here? It looks like the class definition is empty. So, I assume something somewhere is creating an object and using that type, but I don't know where or how. Where would that manifest?

Reply:
Well, that will need some time to explain, hold my beer while I pick the context in other parts of the code

Reply:
OCPError is an pydantic model that also behaves as an exception.
That should not be possible due to metaclasses clashes, but I did an overly complicated thing to make it behave as so.

Let's start with the pretty part:

```python
@raises(SomeException)
class SomePydanticModel(BaseModel):
    ...
```
By wrapping "SomePydanticModel" with "raises" I can make it behave as an exception.
This was how OCPError was created
https://github.com/quipucords/quipucords/blob/1172fedaffe128c35dcf87342292a84e4ca9ec8e/quipucords/scanner/openshift/entities.py#L153-L160

Problem is, OCPError does not get matched to be encoded as json when saving it because `raises` does some dark magic involving, as you probably guessed, `PydanticErrorProxy`. And with this, I can introduce the ugly part:
https://github.com/quipucords/quipucords/blob/1172fedaffe128c35dcf87342292a84e4ca9ec8e/quipucords/compat/pydantic.py#L34-L74

Reply:
TLDR: `PydanticErrorProxy` is the base class for OCPError WHEN it is raised as an exception.

Reply:
Okay, I think I am understanding this more. Thanks for the background!